app/code/Magento/Sales/Controller/AbstractController/Reorder.php

```diff
@@ -57,6 +57,14 @@ public function execute()
         /** @var \Magento\Framework\Controller\Result\Redirect $resultRedirect */
         $resultRedirect = $this->resultRedirectFactory->create();
 
+        $reorderHelper =  $this->_objectManager->get(\Magento\Sales\Helper\Reorder::class);
+        if (!$reorderHelper->canReorder($order->getEntityId())) {
+            $this->messageManager->addErrorMessage(__("Reorder is disabled."));
+            return $resultRedirect->setPath('checkout/cart');
+        }
+
+        $cart = $this->_objectManager->get(\Magento\Checkout\Model\Cart::class);
```

Comment:
I do not think that we need to receive cart twice in the row :)

Reply:
@swnsma - duplicate entry removed.